server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/AbstractEndpoint.java

```diff
@@ -83,6 +97,7 @@ public void badDataType(final Response the_response, final String the_log_messag
   public void unauthorized(final Response the_response, final String the_log_message) {
     the_response.status(HttpStatus.UNAUTHORIZED_401);
     Main.LOGGER.error("unauthorized access 401: " + the_log_message);
+    // TODO: should we halt() the endpoint execution here and simply send the response
```

Comment:
The problem is, you can only `halt()` in the `before` part... not during actual endpoint execution. You can see how I addressed that in the upload endpoints.